core/src/impl/Kokkos_HostThreadTeam.hpp

```diff
@@ -498,7 +498,7 @@ class HostThreadTeamMember {
                                              const int source_team_rank) const
       noexcept {
     KOKKOS_IF_ON_HOST((if (1 < m_data.m_team_size) {
-      T volatile* const shared_value = (T*)m_data.team_reduce();
```

Comment:
In the end, `volatile` was only ever relevant for `Cuda` but this code is never executed in `Cuda` device code. Thus, I would expect that the unit tests would equally pass if we don't use atomics at all. In fact, it seems that `HPX` and `Threads` have their own implementation such that this one would only be used for `OpenMP` and `Serial`.

I agree that using atomics here makes sense, though.

Reply:
If this gets compiled by MSVC, that largely treats `volatile T` as `atomic<T>`